libs/resiliency/tests/unit/async_replay_plain.cpp

```diff
@@ -1,141 +1,75 @@
-        f = hpx::resiliency::experimental::async_replay_validate(
-            10, &validate, &universal_answer);
-        HPX_TEST(f.get() == 42);
-
-        // unsuccessful replay
-        f = hpx::resiliency::experimental::async_replay(6, &deep_thought);
-
-        bool exception_caught = false;
-        try
-        {
-            f.get();
-            HPX_TEST(false);
-        }
-        catch (vogon_exception const&)
-        {
-            exception_caught = true;
-        }
-        catch (...)
-        {
-            HPX_TEST(false);
-        }
-        HPX_TEST(exception_caught);
-
-        // unsuccessful replay validate
-        f = hpx::resiliency::experimental::async_replay_validate(
-            6, &validate, &universal_answer);
-
-        exception_caught = false;
-        try
-        {
-            f.get();
-            HPX_TEST(false);
-        }
-        catch (hpx::resiliency::experimental::abort_replay_exception const&)
-        {
-            exception_caught = true;
-        }
-        catch (...)
-        {
-            HPX_TEST(false);
-        }
-        HPX_TEST(exception_caught);
-
-        // aborted replay
-        f = hpx::resiliency::experimental::async_replay(1, &no_answer);
-
-        exception_caught = false;
-        try
-        {
-            f.get();
-            HPX_TEST(false);
-        }
-        catch (hpx::resiliency::experimental::abort_replay_exception const&)
-        {
-            exception_caught = true;
-        }
-        catch (...)
-        {
-            HPX_TEST(false);
-        }
-        HPX_TEST(exception_caught);
-
-        // aborted replay validate
-        f = hpx::resiliency::experimental::async_replay_validate(
-            1, &validate, &no_answer);
-
-        exception_caught = false;
-        try
-        {
-            f.get();
-        }
-        catch (hpx::resiliency::experimental::abort_replay_exception const&)
-        {
-            exception_caught = true;
-        }
-        catch (...)
-        {
-            HPX_TEST(false);
-        }
-        HPX_TEST(exception_caught);
+            hpx::resiliency::experimental::async_replay(10, &universal_ans);
+
+        auto result = f.get();
+        HPX_TEST(result == 42 || result == 84);
+    }
+
+    {
+        hpx::future<int> f =
+            hpx::resiliency::experimental::async_replay_validate(
+                10, &validate, &universal_ans);
+
+        auto result = f.get();
+        HPX_TEST(result == 42);
+    }
+
+    {
+        universal_action our_action;
+        hpx::future<int> f =
+            hpx::resiliency::experimental::async_replay(locals, our_action);
+
+        auto result = f.get();
+        HPX_TEST(result == 42 || result == 84);
+    }
+
+    {
+        universal_action our_action;
+        hpx::future<int> f =
+            hpx::resiliency::experimental::async_replay_validate(
+                locals, &validate, our_action);
+
+        auto result = f.get();
+
+        HPX_TEST(result == 42);
```

Comment:
What's the rationale of removing the tests?

Reply:
I reduced the complexity of the tests to do exactly what it should do and get result from it. I forgot that CI doesn't run on many localities so I'll have to fix the failing tests.

I'm fine bringing back the old test as well. Thought that the tests should express themselves clearly in cases where they start failing.